src/connector/src/parser/unified/debezium.rs

```diff
@@ -535,9 +540,376 @@ pub fn extract_bson_id(id_type: &DataType, bson_doc: &serde_json::Value) -> Acce
+            if let serde_json::Value::Object(ref obj) = datum
+                && obj.contains_key("$numberLong")
+            {
+                let int_str = obj["$numberLong"].as_str().unwrap_or_default();
+                Some(ScalarImpl::Int256(int_str.parse().unwrap_or_default()))
+            } else {
+                return Err(type_error(datum));
+            }
+        }
+
+        DataType::Float64 => {
+            if let serde_json::Value::Object(ref obj) = datum
+                && obj.contains_key("$numberDouble")
+            {
+                let float_str = obj["$numberDouble"].as_str().unwrap_or_default();
+                let num = match float_str {
+                    "Infinity" => f64::INFINITY,
+                    "-Infinity" => f64::NEG_INFINITY,
+                    "Nan" => f64::NAN,
+                    fstr => fstr.parse().unwrap_or_default(),
+                };
+                Some(ScalarImpl::Float64(num.into()))
+            } else {
+                return Err(type_error(datum));
+            }
+        }
+
+        DataType::Float32 => {
+            if let serde_json::Value::Object(ref obj) = datum
+                && obj.contains_key("$numberDouble")
+            {
+                let float_str = obj["$numberDouble"].as_str().unwrap_or_default();
+                let num = match float_str {
+                    "Infinity" => f32::INFINITY,
+                    "-Infinity" => f32::NEG_INFINITY,
+                    "Nan" => f32::NAN,
+                    fstr => fstr.parse().unwrap_or_default(),
+                };
+                Some(ScalarImpl::Float32(num.into()))
+            } else {
+                return Err(type_error(datum));
+            }
+        }
+
+        DataType::Date | DataType::Timestamp | DataType::Timestamptz => {
+            if let serde_json::Value::Object(ref mp) = datum {
+                if mp.contains_key("$timestamp") {
+                    // todo: fallback to extract date on error
+                    bson_extract_timestamp(datum, type_expected)?
+                } else if mp.contains_key("$date") {
+                    bson_extract_date(datum, type_expected)?
+                } else {
+                    return Err(type_error(datum));
+                }
+            } else {
+                return Err(type_error(datum));
+            }
+        }
+        DataType::Decimal => {
+            if let serde_json::Value::Object(ref obj) = datum
+                && obj.contains_key("$numberDecimal")
+            {
+                let number = obj["$numberDecimal"].as_str().unwrap_or_default();
+                let dec = risingwave_common::types::Decimal::from_str(number).unwrap_or_default();
+                Some(ScalarImpl::Decimal(dec))
+            } else {
+                return Err(type_error(datum));
+            }
+        }
+
+        DataType::Bytea => {
+            if let serde_json::Value::Object(ref obj) = datum
+                && obj.contains_key("$binary")
```

Comment:
The [bson doc](https://www.mongodb.com/docs/manual/reference/mongodb-extended-json/#mongodb-bsontype-Binary) suggests that the binary type is in this format:
```
{ "$binary":
   {
      "base64": "<payload>",
      "subType": "<t>"
   }
}
```

The implementation seems using the wrong format. Am I missing something?